shiny/ui/_chat.py

```diff
@@ -872,6 +873,11 @@ def _trim_messages(
             if remaining_non_system_tokens >= 0:
                 messages2.append(m)
 
+        if format == "anthropic":
+            # For anthropic, the first message must be a user message.
+            while messages2[-1]["role"] != "user":
+                messages2.pop()
```

Comment:
It's technically possible that `messages2` is of length 0 here, so we'll need to be more considerate of that. Also, for similar reasons, it's also possible that there's no user message.

```python
from shiny import reactive
from shiny.express import ui

chat = ui.Chat(id="chat")
chat.ui()

@reactive.effect
def _():
    print(chat.messages(format="anthropic"))
```